dxp/architecture/rich_text_editors.md

```diff
@@ -0,0 +1,208 @@
+| ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | --------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------- |
+| [CKEditorEditor.java](https://github.com/liferay/liferay-portal/blob/a98356e81c2b97c152ee28ab23fcbac8d55bb36d/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/java/com/liferay/frontend/editor/ckeditor/web/internal/CKEditorEditor.java)             | ckeditor        |                                                                                                                                                                                                                         | Default editor for HTML                                                                      |
+| [CKEditorBBCodeEditor.java](https://github.com/liferay/liferay-portal/blob/a98356e81c2b97c152ee28ab23fcbac8d55bb36d/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/java/com/liferay/frontend/editor/ckeditor/web/internal/CKEditorBBCodeEditor.java) | ckeditor_bbcode | [bbcode](https://github.com/liferay/liferay-portal/tree/a98356e81c2b97c152ee28ab23fcbac8d55bb36d/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/bbcode) | Editor with added [BBCode](https://en.wikipedia.org/wiki/BBCode) support for Message Boards  |
+| [CKEditorCreoleEditor.java](https://github.com/liferay/liferay-portal/blob/a98356e81c2b97c152ee28ab23fcbac8d55bb36d/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/java/com/liferay/frontend/editor/ckeditor/web/internal/CKEditorCreoleEditor.java) | ckeditor_creole | [creole](https://github.com/liferay/liferay-portal/tree/a98356e81c2b97c152ee28ab23fcbac8d55bb36d/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/creole) | Editor with added [Creole](<https://en.wikipedia.org/wiki/Creole_(markup)>) support for Wiki |
```

Comment:
About the CI failure. It's because the link-"parser" in the `checkLinks.js` script isn't able to distinguish the end of the URL in`[Creole](<https://en.wikipedia.org/wiki/Creole_(markup)>)`:

```
dxp/architecture/rich_text_editors.md: No heading found matching internal target: #Anatomy
TypeError [ERR_INVALID_URL]: Invalid URL: <https://en.wikipedia.org/wiki/Creole_(markup
```

I don't think there is a good/easy fix for this in the parser, but you can change the link to:

```markdown
[Creole](https://en.wikipedia.org/wiki/Creole_%28markup%29)
```

or use footnote style syntax:

```markdown
[Creole] ← in the table

and then at the footer of the document:

[Creole]: https://en.wikipedia.org/wiki/Creole_(markup)
```

Reply:
Actually, I take back what I said:

> I don't think there is a good/easy fix for this in the parser

I'll send that presently.

But fixing that reveals these:

```
dxp/architecture/rich_text_editors.md: No heading found matching internal target: #Anatomy
dxp/architecture/rich_text_editors.md: No heading found matching internal target: #liferay-editor:resources
dxp/architecture/rich_text_editors.md: No heading found matching internal target: #liferay-editor:editor
```

First one is legit (link is bad). Second and third, I'll fix the link checker.

Reply:
https://github.com/liferay/liferay-frontend-guidelines/pull/142